client/src/main/java/org/apache/rocketmq/client/impl/MQClientAPIImpl.java

```diff
@@ -1697,6 +1697,8 @@ public ClusterInfo getBrokerClusterInfo(
         switch (response.getCode()) {
             case ResponseCode.SUCCESS: {
                 return ClusterInfo.decode(response.getBody(), ClusterInfo.class);
+            } case ResponseCode.BROKER_NOT_EXIST: {
```

Comment:
Maybe this is not an appropriate change, since returning an empty object when an exception is raised would be misleading for the user?

Reply:
@ferrirW  The reason for the modification is that when the broker sends the RemoteingCommand to the Nameserver, the related broker information may not be collected from the NameServer because the netty channel is in the wait state, so an exception is thrown here. Currently only exposed in multi-threaded scenarios. If there is no problem after migrating the test class I will remove this case